src/plugins/intel_gpu/include/intel_gpu/runtime/memory_pool.hpp

```diff
@@ -107,19 +107,22 @@ class memory_pool {
                           const std::set<primitive_id>& restrictions,
                           allocation_type type,
                           bool reusable = true,
-                          bool reset = true);  // get from pool or create memory allocation
+                          bool reset = true,
+                          bool* from_pool = nullptr);  // get from pool or create memory allocation
```

Comment:
Wouldn't it be better to introduce some new field for memory structure? Something like this:
```
struct memory {   
    ...
    std::shared_ptr<MemoryTracker> get_mem_tracker() const { return m_mem_tracker; }

    // Debug memory allocation info
    GPU_DEBUG_CODE(std::string mem_alloc_info);

protected:
    ...
};
```
And set it in `memory_pool::get_from_non_padded_pool` functions as follow:

```
                ...
                auto ret_mem = _engine->reinterpret_buffer(*it->second._memory, layout);
                GPU_DEBUG_CODE(ret_mem->mem_alloc_info = "from_pool");
                return ret_mem;
            }
        }
        auto mem = alloc_memory(layout, type);
        GPU_DEBUG_CODE(mem->mem_alloc_info = "new_alloc");
```
And use this field for statistic collection, just to minimize passing this `from_pool` all over functions call:
`GPU_DEBUG_PROFILED_STAGE_MEMALLOC_INFO(memory->mem_alloc_info);`

Reply:
Thats a good suggestion, I will apply it. Thanks :)

Reply:
I updated PR. Please review!